traits-stubs/traits_stubs_tests/numpy_examples/Array.py

```diff
@@ -0,0 +1,28 @@
+# (C) Copyright 2005-2022 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+import numpy as np
+
+from traits.api import Array, ArrayOrNone, HasTraits
+
+
+class HasArrayTraits(HasTraits):
+    spectrum = Array(shape=(None,), dtype=np.float64)
+    maybe_image = ArrayOrNone(shape=(None, None, 3), dtype=np.float64)
```

Comment:
Do we want a test for `CArray`?

Similarly, it might be good to test the dimension ranges (eg. `maybe_rgba = Array(shape=(None, None, (3,4)), ...)

Also, is it worth testing for invalid trait parameters (eg. passing a default value of `None` to `Array`)?

Some of this may be overkill if we expect that things will Just Work.

Reply:
We should probably test `CArray`. I'm a bit grumpy with `CArray`, because at this point it's actually identical in functionality to `Array`, so I'm thinking that we might want to deprecate it.

I can flesh out the tests a bit more to test invalid parameters.

As an aside, I think at some point we're going to need a better framework for these tests; right now, there's a lot of interaction between tests that should be independent of one another.

Reply:
I've added some tests.

Reply:
> eg. passing a default value of None to Array

That particular example is troublesome: as far as I can tell, in general there's no way to type hint a function parameter that has a default value of `None`, but for which the intended API is that `None` is never passed explicitly.

I've added a few more tests, including tests for bad declarations.